clients/python/girder_client/__init__.py

```diff
@@ -123,8 +138,13 @@ class GirderClient(object):
     # The current maximum chunk size for uploading file chunks
     MAX_CHUNK_SIZE = 1024 * 1024 * 64
 
+    DEFAULT_API_ROOT = 'api/v1'
+    DEFAULT_HOST = 'localhost'
+    DEFAULT_PORT = 443
+    DEFAULT_SCHEME = 'https'
+
     def __init__(self, host=None, port=None, apiRoot=None, scheme=None, apiUrl=None,
-                 cacheSettings=None):
+                 cacheSettings=None, progressReporterCls=DownloadProgressReporter):
```

Comment:
Let's make this default to `None` to avoid mutable default, and use this as the default internally. Also, this parameter needs a docstring.